executor/executor.go

```diff
@@ -170,8 +170,6 @@ func (e *Executor) Run(keys state.Keys, f func() error) {
 			if v == state.Read {
 				// If we don't need exclusive access to a key, just mark
 				// that we are reading it and that we are a reader of it.
-				// TODO: currently a no-op when we add ourself multiple times
-				// but find a better way of handling [reading]
 				t.reading[lt.id] = lt
```

Comment:
I think we should note clearly the scenario for why we use a map here (i.e. that a single task can have different keys that are all just readers of another task)